### ISSUE TITLE
Not working on new IOS updated

### ISSUE THREAD

**Issue by m1narulal1** [pichillilorenzo/flutter_inappwebview#2283]
- [x] I have read the [Getting Started](https://inappwebview.dev/docs/intro/) section
- [x] I have already searched for the same problem

## Environment

| Technology            | Version       |
|-----------------------| ------------- |
| Flutter version       |   3.24.3      |
| Plugin version        |   6.0.0        |
| Android version       |               |
| iOS version           |        18.0       |
| macOS version         |               |
| Xcode version         |               |
| Google Chrome version |               |

Device information: 

## Description

**Expected behavior:** 

**Current behavior:** 

## Steps to reproduce

1. This
2. Than that
3. Then

## Stacktrace/Logcat   
Swift Compiler Error (Xcode): Method does not override any method from its superclass
/Users/technoexponent/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift:1431:25

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/technoexponent/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift:2859:22

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/technoexponent/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift:2870:22

**Comment by MrLightful** [comment]
Yep, same.

**Comment by arrrrny** [comment]
You can use my new package [zikzak_inappwebview](https://pub.dev/packages/zikzak_inappwebview) fork of this package that I will keep maintaining. Thanks @pichillilorenzo for developing/maintaining this package for a long time. I guess he is busy or on vacataion. I am the builder of Flutter Vendure (open source headless commerce framework) package. [Vendure](https://vendure.io/) Please open your PRs [here](https://github.com/arrrrny/zikzak_inappwebview) and I am happy to check them and merge them by Monday.

**Comment by hunguesthotels** [comment]
Same error, save versions.

IOS: 18
Flutter: 3.24.3
Dart 3.5.3
Xcode: 16.0 (16A242d)
Pl
![Képernyőfotó 2024-09-20 - 13 02 30](https://github.com/user-attachments/assets/03694338-2e50-4b5a-9631-cf306e0a9b0b)
ugin: 6.0.0

![Képernyőfotó 2024-09-20 - 12 49 37](https://github.com/user-attachments/assets/bf074059-49f0-47bf-abd6-7f0d0e065860)
![Képernyőfotó 2024-09-20 - 12 49 23](https://github.com/user-attachments/assets/4bd13e55-f3ec-43ab-b899-6ebbb2ec52b9)

**Comment by nemscep** [comment]
This is an issue with xcode 16. 
What worked for me was to install previous version of xcode 15.4 using [xcodes](https://formulae.brew.sh/cask/xcodes) and then changing command line tools version in xcode locations settings to 15.4 instead of 16 and voila, everything is back to working again 😄

**Comment by HarshuPatel007** [comment]
Hello, folks 
I found a temporary solution for the issue in Xcode. In the `InAppWebView` class under the iOS folder, you need to replace the following code in the Java file when you encounter the error:

**Replace:**
```swift
public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: ((Any?, Error?) -> Void)? = nil)
```
**With:**
```swift
open override func evaluateJavaScript(_ javaScriptString: String, completionHandler: (@MainActor (Any?, (any Error)?) -> Void)? = nil)
```

Additionally, I was facing package-related errors, so I updated the Firebase packages according to my Flutter version. Now, the app is finally running on iOS 18.

After that, I encountered a "command non-zero exit" error. The solution for this is to make changes within the **Build Settings** of the Runner project, as shown in the screenshot below.

![Screenshot_1](https://github.com/user-attachments/assets/88ad7418-30b3-45f6-8fb8-f5ec90f6d0d7)

**Comment by amit-iconflux** [comment]
+1

**Comment by shivraj-walke-bbi** [comment]
+1

**Comment by shivraj-walke-bbi** [comment]
> Hello, folks I found a temporary solution for the issue in Xcode. In the `InAppWebView` class under the iOS folder, you need to replace the following code in the Java file when you encounter the error:
> 
> **Replace:**
> 
> ```swift
> public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: ((Any?, Error?) -> Void)? = nil)
> ```
> 
> **With:**
> 
> ```swift
> open override func evaluateJavaScript(_ javaScriptString: String, completionHandler: (@MainActor (Any?, (any Error)?) -> Void)? = nil)
> ```
> 
> Additionally, I was facing package-related errors, so I updated the Firebase packages according to my Flutter version. Now, the app is finally running on iOS 18.
> 
> After that, I encountered a "command non-zero exit" error. The solution for this is to make changes within the **Build Settings** of the Runner project, as shown in the screenshot below.
> 
> ![Screenshot_1](https://private-user-images.githubusercontent.com/81405835/369778583-88ad7418-30b3-45f6-8fb8-f5ec90f6d0d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwOTAyNDksIm5iZiI6MTcyNzA4OTk0OSwicGF0aCI6Ii84MTQwNTgzNS8zNjk3Nzg1ODMtODhhZDc0MTgtMzBiMy00NWY2LThmYjgtZjVlYzkwZjZkMGQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDExMTIyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkZTU2ZWQyNzI2ZjgxYmEwZDMwNDcyN2MwNzA4NjVhN2QyYTQzNDk5ZDg0ODBiZGRiZGJmYWM2YWE4Nzk0YTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uCr6J7cZoAapYY-MgrxNclFkxUkmBH-okqEWVV_Gsto)
> 
> Thank you.



This Not working. for me

**Comment by plusema86** [comment]
> This is an issue with xcode 16. What worked for me was to install previous version of xcode 15.4 using [xcodes](https://formulae.brew.sh/cask/xcodes) and then changing command line tools version in xcode locations settings to 15.4 instead of 16 and voila, everything is back to working again 😄

This worked for me, but since I upgraded MacOS to Sequoia, I had to perform these additional steps

- open a terminal
- open folder where Xcode is installed (Applications)
- drag and drop the Xcode app into the terminal
- append /Contents/MacOS/Xcode
- run the command

**Comment by pichillilorenzo** [comment]
Released new version 6.1.0 with the fix, thanks.

**Comment by felipecastrosales** [comment]
Hello everyone! I opened an issue https://github.com/pichillilorenzo/flutter_inappwebview/issues/2327 related to a new problem like this. I ask that you consider upvoting or interacting so that the maintainers can see this as soon as possible. Remember that these problems can happen to you - as is happening to me. :)
